tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -98,6 +138,48 @@ private static void testKeyTypeMismatch(final String algorithm, final String bas
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("ecdsaBadSignatures")
+    @SuppressWarnings("unchecked")
+    public void testBadEcdsaSignature(final String algorithm, final BigInteger r, final BigInteger s)
+            throws GeneralSecurityException {
+        final Signature verifier = Signature.getInstance(algorithm, NATIVE_PROVIDER);
+        // We always use the largest key to avoid hash truncation confusion
+        final ECPublicKey pubKey = (ECPublicKey) ECDSA_PAIR.getPublic();
+        final byte[] signature;
+        byte[] rArr = r.toByteArray();
+        if (rArr[0] == 0) { // Trim leading zero byte if present
+            rArr = Arrays.copyOfRange(rArr, 1, rArr.length);
+        }
+        byte[] sArr = s.toByteArray();
+        if (sArr[0] == 0) { // Trim leading zero byte if present
+            sArr = Arrays.copyOfRange(sArr, 1, sArr.length);
+        }
+
+        if (algorithm.endsWith("inP1363Format")) {
+            // Just zeros of the appropriate length
+            final int elementLength = (pubKey.getParams().getOrder().bitLength() + 7) / 8;
+            signature = new byte[elementLength * 2];
+            System.arraycopy(rArr, 0, signature, elementLength - rArr.length, rArr.length);
+            System.arraycopy(sArr, 0, signature, 2 * elementLength - sArr.length, sArr.length);
+        } else {
+            // It isn't worth building full ASN.1 logic here, so we cover the easy cases of ones and zeros
```

Comment:
doesn't BouncyCastle have [facilities](https://people.eecs.berkeley.edu/~jonah/bc/org/bouncycastle/asn1/package-summary.html) for this?

Reply:
Yeah, and we do have a dependency of BC for our tests, but I really just use it for cryptography. The math is the same between these two algorithms and only the signature parsing varies. I don't think it is worth the complexity here.

Reply:
fair enough.